### DISCUSSION TITLE
[Scheduler?] How to dynamically add schedules

### DISCUSSION THREAD

**Discussion by Gerben321** [symfony/symfony#54139] [Q&A]
Hi all,

I'm working on an app on which users can set schedules. For example after certain events they should get a message that x and y happened.

Now my next type of schedule is a time interval. So users can set a time (planning to use the DateIntervalType) after which they should be notified that the schedule is overdue. I was looking for ways to solve this is a good way. The messenger component clearly wasn't good for it. So I found out about the scheduler component. It looks and sounds so great, it even talks about 'dynamic' stuff.

So I tried it out, but it clearly became visible this was not dynamic at all. All you can do is set a predetermined interval after which something should happen. This really does not work for me.

What I am looking for is something that I can schedule. So it has to be dynamic. A user can either set it to 1,2,3 days, 2 months, anything. That next moment in time should trigger something so I can alert the users that it's time for the schedule. Once the user have acknowledged the schedule, only then should it start again at the previously given interval. So it's like a periodic interval, but should not repeat until the user has marked it as done.

Also if the user marks it as done before the end, it should remove the old schedule and create a new one with the interval given previously, but added at the moment the user marked it done.

I'm afraid the scheduler component isn't a the right fit for me, although it sounds good. Anyone that can point me in the right direction? I'm on a host with cronjobs only, so no fancy deamons and such.

Thanks!

**Comment by MatTheCat** [comment]
Hello,

AFAIK you cannot indeed change a schedule’s period at runtime, but you can work around this by generating messages yourself using a [`CallbackMessageProvider`](https://symfony.com/doc/current/scheduler.html#a-dynamic-vision-for-the-messages-generated).

As I’m not entirely sure to understand your use-case, maybe check [delayed messages](https://symfony.com/doc/current/messenger.html#envelopes-stamps) too?

**Comment by vBruny** [comment]
Hello, 

What if you create your own Trigger? 

You can make a first message that trigger according to your DateInterval and alert user, then a second message that trigger only if the user have acknowledged the schedule. Both triggers could be custom with simple logic like read boolean or interval in DB then check the actual time.

For the "remove before done" feature, I think you can do this with [add and remove method in your schedule](https://symfony.com/doc/current/scheduler.html#strategies-for-adding-removing-and-modifying-entries-within-the-schedule)

Hope it is what you are looking for.
